source/extensions/quic_listeners/quiche/envoy_quic_proof_source.cc

```diff
@@ -20,20 +20,19 @@ EnvoyQuicProofSource::GetCertChain(const quic::QuicSocketAddress& server_address
                                    const std::string& hostname) {
   absl::optional<std::reference_wrapper<const Envoy::Ssl::TlsCertificateConfig>> cert_config_ref =
       getTlsCertConfig(server_address, client_address, hostname);
-  // Only return the first TLS cert config.
   if (!cert_config_ref.has_value()) {
     ENVOY_LOG(warn, "No matching filter chain found for handshake.");
-    return quic::QuicReferenceCountedPointer<quic::ProofSource::Chain>(
-        new quic::ProofSource::Chain({}));
+    return nullptr;
   }
   auto& cert_config = cert_config_ref.value().get();
   const std::string& chain_str = cert_config.certificateChain();
   std::string pem_str = std::string(const_cast<char*>(chain_str.data()), chain_str.size());
   std::stringstream pem_stream(chain_str);
   std::vector<std::string> chain = quic::CertificateView::LoadPemFromStream(&pem_stream);
   if (chain.empty()) {
-    throw EnvoyException(
-        absl::StrCat("Failed to load certificate chain from ", cert_config.certificateChainPath()));
+    ENVOY_LOG(warn, "Failed to load certificate chain from %s", cert_config.certificateChainPath());
+    return quic::QuicReferenceCountedPointer<quic::ProofSource::Chain>(
+        new quic::ProofSource::Chain({}));
```

Comment:
What's the difference between returning a nullptr and returning a non-nullptr but the chain is empty? Does caller need to handle them differently?